Gems/ROS2/Code/Source/Camera/CameraSensorDescription.cpp

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include "CameraSensorDescription.h"
+#include <AzCore/Math/MatrixUtils.h>
+
+namespace ROS2
+{
+    CameraSensorDescription::CameraSensorDescription(
+        const AZStd::string& cameraName, const CameraConfiguration& configuration, const SensorConfiguration& sensorConfiguration)
+        : m_cameraConfiguration(configuration)
+        , m_sensorConfiguration(sensorConfiguration)
+        , m_cameraName(cameraName)
+        , m_aspectRatio(static_cast<float>(configuration.m_width) / static_cast<float>(configuration.m_height))
+        , m_viewToClipMatrix(MakeViewToClipMatrix())
+        , m_cameraIntrinsics(MakeCameraIntrinsics())
+    {
+        ValidateParameters();
```

Comment:
I'm thinking maybe we should `ValidateParameters` on the arguments first before assignment, instead of assigning through the initializer list. We may be able to validate the divide by zero check mentioned above that way